src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+  let prevState = { ...state };
+
+  for (const action of actions) {
+    let newState = { ...prevState };
+
+    switch (action.type) {
```

Comment:
You can use destructurization here for action
```suggestion
  for (const action of actions) {
    let newState = { ...prevState };
    const {} = action;

    switch (action.type) {
```